aws/src/test/java/org/apache/iceberg/aws/glue/InMemoryLockManagerTest.java

```diff
@@ -0,0 +1,152 @@
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InMemoryLockManagerTest {
+
+  private LockManagers.InMemoryLockManager lockManager;
+  private String lockEntityId;
+  private String ownerId;
+
+  @Before
+  public void before() {
+    lockEntityId = UUID.randomUUID().toString();
+    ownerId = UUID.randomUUID().toString();
+    lockManager = new LockManagers.InMemoryLockManager(Maps.newHashMap());
+  }
+
+  @After
+  public void after() {
+    lockManager.close();
+  }
+
+  @Test
+  public void testAcquireOnce_singleProcess() {
+    lockManager.acquireOnce(lockEntityId, ownerId);
+  }
+
+  @Test
+  public void testAcquireOnce_multiProcess() {
+    List<Boolean> results = IntStream.range(0, 10).parallel()
+        .mapToObj(i -> {
+          try {
+            lockManager.acquireOnce(lockEntityId, ownerId);
+            return true;
+          } catch (IllegalStateException e) {
+            return false;
+          }
+        })
+        .collect(Collectors.toList());
+    Assert.assertEquals(
+        "only 1 thread should have acquired the lock",
+        1, results.stream().filter(s -> s).count());
+  }
+
+  @Test
+  public void testReleaseAndAcquire() {
+    Assert.assertTrue(lockManager.acquire(lockEntityId, ownerId));
+    lockManager.release(lockEntityId, ownerId);
+    Assert.assertTrue(lockManager.acquire(lockEntityId, ownerId));
+  }
+
+  @Test
+  public void testReleaseWithWrongOwner() {
+    Assert.assertTrue(lockManager.acquire(lockEntityId, ownerId));
+    AssertHelpers.assertThrows("should throw exception if ownerId is wrong",
+        IllegalArgumentException.class,
+        "current owner",
+        () -> lockManager.release(lockEntityId, UUID.randomUUID().toString()));
+  }
+
+  @Test
+  public void testAcquire_singleProcess() throws Exception {
+    Assert.assertTrue(lockManager.acquire(lockEntityId, ownerId));
+    String oldOwner = ownerId;
+
+    CompletableFuture.supplyAsync(() -> {
+      try {
+        Thread.sleep(5000);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+      lockManager.release(lockEntityId, oldOwner);
+      return null;
+    });
+
+    ownerId = UUID.randomUUID().toString();
+    long start = System.currentTimeMillis();
+    Assert.assertTrue(lockManager.acquire(lockEntityId, ownerId));
+    Assert.assertTrue("should succeed after 5 seconds",
+        System.currentTimeMillis() - start >= 5000);
```

Comment:
Can you make this interval shorter? 5s is a long time to wait. 200ms would do fine.

Reply:
Updated